pkg/graveler/graveler.go

```diff
@@ -891,11 +934,117 @@ func (g *Graveler) GetTag(ctx context.Context, repositoryID RepositoryID, tagID
 }
 
 func (g *Graveler) CreateTag(ctx context.Context, repositoryID RepositoryID, tagID TagID, commitID CommitID) error {
-	return g.RefManager.CreateTag(ctx, repositoryID, tagID, commitID)
+	// Check that Tag doesn't exist before running hook
```

Comment:
Just to make the API more expected, fetch the repo first - so you will get repository not found and not tag not found.